lib/collection/src/shards/replica_set.rs

```diff
@@ -1058,6 +1068,9 @@ impl ShardReplicaSet {
                 Some(ReplicaState::Initializing) => {
                     Ok(Some(local_shard.get().update(operation, wait).await?))
                 }
+                Some(ReplicaState::Listener) => {
+                    Ok(Some(local_shard.get().update(operation, false).await?))
```

Comment:
Do we want listener nodes to have zero impact on the update requests?

If so we might want to catch the potential errors returned by `update`.
Most likely issues when interacting with the `wal`.

Reply:
It doesn't guarantee to have zero impact, but the impact is less than regular nodes